online-shop/src/app/product-list/product-list.component.ts

```diff
@@ -14,11 +16,15 @@ import { ProductService } from '../service/product.service';
 })
 
 export class ProductListComponent implements OnInit {
-  products$: Observable<IProduct[]>;
-
-  constructor(private service: ProductService){}
+  dataSource: any[]
+  public allProducts$ = this.store.select(selectAllProducts);
+  columndefs : any[] = ['name','category', 'price','id'];
+  constructor(private store: Store<AppState> ){}
   ngOnInit() {
-    this.products$ = this.service.getAllProducts()
+
+    this.store.dispatch(loadProducts());
+    this.allProducts$.subscribe((data) => {this.dataSource = data})
```

Comment:
It might be a good practice to `.unsubscribe()` in an `ngOnDestroy` component lifecycle, in order to avoid memory leaks in your code: https://www.thisdot.co/blog/best-practices-for-managing-rxjs-subscriptions/